Model/RSCRecord.php

```diff
@@ -166,47 +155,53 @@ public function save($data = null, $validate = true, $fieldList = array()) {
 		}
 		$zone = $data['zone'];
 		unset($data['zone']);
-		if ($this->recordExists($data['name'], $zone)) {
-			$Record = $this->__getRecordByNameAndZone($data['name'], $zone);
+		$type = empty($data['type']) ? 'CNAME' : $data['type'];
+		$Record = $this->__getRecordByNameZoneType($data['name'], $zone, $type);
+		if (!empty($Record)) {
 			$Record->update($data);
-		} else {
-			$Domain = $this->__getDomainObjectByZone($zone);
-			$Record = $Domain->Record()->Create($data);
+		} else {  // Create it
+			$async = $this->__getDomainObjectByZone($zone)->record($data)->create();
+			$async->waitFor('COMPLETED');  // Wait for asyncresponse to complete or error out
+			if ($async->status == 'ERROR') {
+				$this->_error(!empty($async->error->details) ? $async->error->details : "Unable to create DNS record for {$data['name']} on zone {$zone}.");
+			}
+			$Record = !empty($async->response->records) ? $async->response->records[0] : null;
 		}
 		if ($Record) {
-			return array(
-				$this->alias => array(
-					'id' => $Record->Id(),
-					'name' => $Record->Name(),
+			return [
+				$this->alias => [
+					'id' => $Record->id,
+					'name' => $Record->name,
 					'zone' => $zone,
 					'type' => $Record->type,
 					'ttl' => $Record->ttl,
 					'data' => $Record->data,
 					'created' => $Record->created,
 					'updated' => $Record->updated,
-				)
-			);
+				]
+			];
 		}
 		return false;
 	}
 	
 	/**
 	* Delete the Domain
-	* @param string name
-	* @param string zone
+	* @param string $name
+	* @param string $zone
+	* @param string $type (optional)
 	* @return boolean success
 	*/
-	public function delete($name = null, $zone = true /* true is for strict compliance */) {
+	public function delete($name = null, $zone = true, $type = 'CNAME') {  // true is for strict compliance
 		if (!$this->DNS) {
-			$this->_erorr('Unable to connect to DNS.');
+			$this->_error('Unable to connect to DNS.');
 		}
 		if (empty($name) || $zone === true || empty($zone)) {
 			$this->_error('zone and name are required to delete a record');
 		}
-		if (!$this->recordExists($name, $zone)) {
-			$this->_erorr("$name doesn't exist on $zone DNS.");
+		$Record = $this->__getRecordByNameZoneType($name, $zone, $type);
```

Comment:
Small nitpick about comments/annotations (that existed before your changes so not even about your code really):

Should the $zone type be changed from `string` to `mixed` or `string|true`, considering the default is boolean `true` and there is a `$zone === true` check in the method? You could move that "true is for strict compliance " comment to the $zone annotation as well:

`* @param string|true $zone Zone name; TRUE (default) is for strict compliance`

Reply:
I would change this but honestly this whole package is full of docblock errors, that it's not really worth it to go through and fix it right now, and update all the references, re-merge, re-push, etc.